invenio_vocabularies/contrib/subjects/schema.py

```diff
@@ -23,13 +33,33 @@ class SubjectSchema(BaseVocabularySchema):
     # here.
     id = SanitizedUnicode(required=True)
     scheme = SanitizedUnicode(required=True)
-    subject = SanitizedUnicode(required=True)
+    subject = i18n_strings
+    synonyms = fields.List(SanitizedUnicode())
+    identifiers = IdentifierSet(
+        fields.Nested(
+            partial(
+                IdentifierSchema,
+                allowed_schemes=subject_schemes,
+                identifier_required=False,
+            )
+        )
+    )
```

Comment:
Hmmm... I think I lack the context for this change. 

My initial reaction would be:

- yes for the idea of capturing the translated labels
- no for almost anything else because too much change at once :upside_down_face: 

Detail
- Why synonyms? With 1M+ subjects for us, it's already near impossible to surface the right one with current UI interface. If different UI interface perhaps... but if loosey (non-controlled vocab), then let users use keywords. The amount of data and sourcing for this makes me weary too...
- identifiers: are those equivalent identifiers in other vocabularies? What source for these? If one has LCSH and GND subjects, do both hold that information? this duplicates data tremendously (just with 2 subjects we would get at least 2x usage that grows exponentially from there...) + how to keep in sync. If no duplication, which is the main subject?

My take:

If we do want this additional data, I think we need to extract the concept represented by these additions into its own thing: a new resource called "SubjectEquivalencies" or "SubjectSets" where each entry captures that locus of equivalent subjects/keywords. No duplication, clear access, clear concept. But again this might be much for the original task to just move to datastreams.

Reply:
Synonyms are useful in case of abbreviations which are not of subject headings on their own (e.g. "European Union" vs. "EU"). As you never know which exact term is part of the subject headings you have a better chance to find it by synonyms (if available) or translations. So the key is: it's not more subjects but just better chances to find the correct one in case you don't know the exact term.
Regarding the identifiers, perhaps it's just overdone.